db/events.php

```diff
@@ -36,10 +36,14 @@
     ],
     [
         'eventname' => '\core\event\course_module_created',
-        'callback' => '\mod_learningmap\autoupdate::reset_backlink_cache',
+        'callback' => '\mod_learningmap\autoupdate::reset_backlink_cache_if_necessary',
     ],
     [
         'eventname' => '\core\event\course_module_updated',
-        'callback' => '\mod_learningmap\autoupdate::reset_backlink_cache',
+        'callback' => '\mod_learningmap\autoupdate::reset_backlink_cache_if_necessary',
+    ],
+    [
```

Comment:
Please add a comment that this is needed in case the course format changed. Maybe it would also be nice to make a separate callback function handling this event because it should be possible to extract if the event actually was caused by a course format change?

Currently the cache is rebuilt on every course update, right?